CHANGES.rst

```diff
@@ -2,11 +2,18 @@
 CHANGES
 =======
 
-3.6.5 (unreleased)
+4.0.0 (unreleased)
 ------------------
 
-- Nothing changed yet.
+- Add support for Python 3.5, 3.6 and PyPy.
 
+- Replace dependency on ``ZODB3`` with ``BTrees`` and ``persistent``.
+
+- Drop test dependency on ``zope.app.testing``.
+
+- The synchronization view now uses Python's built-in transport for
+  handling Basic Authentication. This means that the username and
+  password should not contain a ``:``.
```

Comment:
I don't see that the previous implementation actually did any special handling of colons for either the username or password, so I don't think this is a regression:
```python
        if self.username is not None and self.password is not None:
             h.putheader("AUTHORIZATION", "Basic %s" %
                         encodestring("%s:%s" % (self.username, self.password)
                                      ).replace("\012", ""))
```

The [generic URI syntax](https://tools.ietf.org/html/rfc3986#section-3.2.1) doesn't specify any special standard handling for ':'.

The [HTTP Basic Auth RFC](https://tools.ietf.org/html/rfc7617#section-2) states:

>Furthermore, a user-id containing a colon character is invalid, as
   the first colon in a user-pass string separates user-id and password
   from one another; text after the first colon is part of the password.
   User-ids containing colons cannot be encoded in user-pass strings.

>   Note that many user agents produce user-pass strings without checking
   that user-ids supplied by users do not contain colons; recipients
   will then treat part of the username input as part of the password.

I think we (still) comply with that.

So it looks like the change note just needs to be updated a bit: colons were never allowed in passwords.